packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -183,7 +190,30 @@ class ${class_name}:
             Stream of measurement output.
         """
         parameter_values = [${measure_api_parameters}]
-        request = self._create_measure_request(parameter_values)
-
-        for response in self._get_stub().Measure(request):
-            yield self._deserialize_response(response)
+        with self._initialization_lock:
+            if self._measure_response is not None:
+                raise RuntimeError("A measure call is already in progress.")
+            request = self._create_measure_request(parameter_values)
+            self._measure_response = self._get_stub().Measure(request)
+
+        try:
+            for response in self._measure_response:
+                % if output_metadata:
+                yield self._deserialize_response(response)
+                % else:
+                yield
+                % endif
+        except grpc.RpcError as e:
+            if e.code() == grpc.StatusCode.CANCELLED:
+                print("The measure call is canceled.")
+            else:
+                raise
+        finally:
+            self._measure_response = None
+
+    def cancel(self) -> bool:
+        if self._measure_response and self._measure_response.is_active():
+            self._measure_response.cancel()
+            return True
```

Comment:
This has two potential race conditions. 
- The RPC may complete between the call to `is_active()` and `cancel()`.
- `self._measure_response` may be set to `None` between the call to `is_active()` and `cancel()`.

Do you really need to call `is_active()`?

Does the underlying future return a bool? I think it does.

Try something like this:
```
response = self._measure_response
if response:
    return response.cancel()
return False
```